src/modules/ekf2/EKF/gps_control.cpp

```diff
@@ -83,15 +81,17 @@ void Ekf::controlGpsFusion()
 		const float pos_var = sq(pos_noise);
 		const Vector2f pos_obs_var(pos_var, pos_var);
 		updateHorizontalPositionAidSrcStatus(gps_sample.time_us,
-						     gps_sample.pos,                             // observation
+						     position,                                   // observation
 						     pos_obs_var,                                // observation variance
 						     math::max(_params.gps_pos_innov_gate, 1.f), // innovation gate
 						     _aid_src_gnss_pos);
 		_aid_src_gnss_pos.fusion_enabled = (_params.gnss_ctrl & GnssCtrl::HPOS);
 
 		// update GSF yaw estimator velocity (basic sanity check on GNSS velocity data)
-		if (gps_checks_passing && !gps_checks_failing) {
-			_yawEstimator.setVelocity(gps_sample.vel.xy(), gps_sample.sacc);
+		if (_gps_speed_valid && velocity.isAllFinite()
```

Comment:
Might as well respect EKF2_GPS_CTRL velocity here.

```suggestion
		if ((_params.gnss_ctrl & GnssCtrl::VEL) && _gps_speed_valid && velocity.isAllFinite()
```